quconot/implementations/mct_no_auxiliary_relative.py

```diff
@@ -1,79 +1,79 @@
-# Quconot/quconot/implementations/mct_no_auxiliary_relative_phase.py
-#
-# Authors:
-#  - Handy Kurniawan
-#
-# Apply the implementation from Qiskit MCT
-
-from copy import deepcopy
-from typing import List
-
-from qiskit import QuantumCircuit, transpile
-
-from .mct_base import MCTBase
-
-
-class MCTNoAuxiliaryRelativePhase(MCTBase):
-    def __init__(self, controls_no: int, **kwargs) -> None:
-        assert (
-            controls_no >= 2 and controls_no <= 3
-        ), "At the moment we cannot handle controls bigger than 3."
-        self._n = controls_no
-        self._circuit: QuantumCircuit = None
-        pass
-
-    @classmethod
-    def verify_mct_cases(
-        self,
-        controls_no: int,
-        max_auxiliary: int,
-        relative_phase: bool = False,
-        clean_acilla: bool = True,
-        wasted_auxiliary: bool = False,
-        separable_wasted_auxiliary: bool = False,
-    ) -> List["MCTBase"]:
-        """Generate all possible MCT implementation satisfying the requirements
-
-        relative_phase: true / false (D)
-        clean_auxiliary: true (D) / false
-        wasted_auxiliary: true / false (D)
-        separable_wasted_auxiliary: true / false (D)    # requires wasted_auxiliary set to True
-
-        :return: a quantum circuit
-        :rtype: QuantumCircuit
-        """
-        return [MCTNoAuxiliaryRelativePhase(controls_no)]
-
-    def generate_circuit(self) -> QuantumCircuit:
-        """Return a QuantumCircuit implementation
-
-        :return: a quantum circuit
-        :rtype: QuantumCircuit
-        """
-        qc = QuantumCircuit(self._n + 1)
-        if self._n == 2:
-            qc.rccx(0, 1, 2)
-        elif self._n == 3:
-            qc.rcccx(0, 1, 2, 3)
-        else:
-            raise ValueError("At the moment we cannot handle controls bigger than 3.")
-
-        # should be done for all implementations
-        # TODO: solve issue with reordered qubits
-        self._circuit = transpile(qc, basis_gates=["cx", "u3"])
-        return deepcopy(self._circuit)
-
-    def num_auxiliary_qubits(self):
-        """Return number of auxiliary qubits
-
-        :return: number of auxiliary qubits
-        :rtype: int
-        """
-        return 0
-
-
-if __name__ == "__main__":
-    mct = MCTNoAuxiliaryRelativePhase(4)
-
-    circ = mct.generate_circuit()
-    # print(circ.draw())
+# Quconot/quconot/implementations/mct_no_auxiliary_relative_phase.py
+#
+# Authors:
+#  - Handy Kurniawan
+#
+# Apply the implementation from Qiskit MCT
+
+from copy import deepcopy
+from typing import List
+
+from qiskit import QuantumCircuit, transpile
+
+from .mct_base import MCTBase
+
+
+class MCTNoAuxiliaryRelative(MCTBase):
```

Comment:
the name should relate to qiskit, or original paper if applicable

Reply:
From qiskit i am using the rccx and rcccx for this implementation.

Reply:
![image](https://user-images.githubusercontent.com/62504305/212589398-8306147b-3e29-4234-886a-7d986a42101e.png)

For `rccx`

Reply:
![image](https://user-images.githubusercontent.com/62504305/212589478-2c46d8c3-b901-461a-b527-df25a57bd274.png)

For `rcccx`

Reply:
Based on the above paper, there are also more implementations:

![image](https://user-images.githubusercontent.com/62504305/212589645-98d30fae-097d-4a26-bb8b-c176ffafaac5.png)

Reply:
how about these names:
- `mct_qiskit_no_ancilla`
- `mct_qiskit_rccx`
- `mct_qiskit_rcccx`
- `mct_qiskit_recurssion`
- `mct_qiskit_vchain`
- `mct_qiskit_vchain_dirty`

Reply:
`MaslovSmallCircuits`?
@ankit27kh perhaps you could help with naming?

Reply:
@adamglos92 @HandyKurniawan 
changing the names of the implementations is something I have thought of earlier. I would want all implementations to have new names.
I suggest we open an issue to consider the naming issue and not hold this PR back.

Reply:
agree!

Reply:
#12 
created an issue so we remember this